doc/quick_start/contributor-guide.rst

```diff
@@ -0,0 +1,277 @@
+
+#. Once the bug is fixed upstream, the feature owner owns merging the upstream code back into Ostro OS. 
+#. QA will verify the fix with the Ostro image and update the Ostro Project bug status properly, and track the issue until it's closed.
+
+The upstream developer and QA owner take responsibility to drive a fix. Based on the bug's impact and available resources, 
+the Ostro feature owner decides whether a hotfix is required in Ostro OS before the upstream project releases a fixed version. The
+feature owner is also responsible for removing the hotfix when appropriate.
+
+When submitting a new bug report (or feature request) to Ostro Project or an upstream project, verify 
+that the issue has not been reported already to avoid duplication. 
+
+
+Submitting Patches to the Ostro Project
+=======================================
+
+For general patches to upstream packages, we recommend you submit them directly to the
+appropriate upstream project home. For patches specific to the Ostro OS, submit them as a 
+git pull request (PR) so project maintainers can review and merge them. In this section we'll 
+explain how to properly format and submit your patch.
+
+In a collaborative open source environment, standards and methods for submitting changes 
+help reduce chaos that can result from an active development community. One general practice 
+is to make small, controlled changes. This practice simplifies review, makes merging and 
+rebasing easier, and keeps the change history clear and clean.
+
+The Ostro OS is based on Yocto Project layers that are put together using the combo-layer script.
+The http://github.com/ostroproject/ostro-os repository is a combination of several components in a 
+single repository and contains everything needed to build Ostro OS, including:
+
+- bitbake
+- openembedded-core
+- meta-intel
+- meta-ostro
+- meta-ostro-fixes
+- meta-ostro-bsp
+- meta-intel-iot-security
+- meta-appfw
+- meta-openembedded
+- meta-oic
+- meta-intel-iot-middleware
+- meta-iotqa
+- meta-iot-web
+- meta-security-isafw
+- meta-yocto
+- meta-java
+- meta-soletta
+- meta-swupd
+
+.. _`meta-ostro README`: https://github.com/ostroproject/meta-ostro/blob/master/README.rst
+
+See the `meta-ostro README`_ for an up-to-date list on the GitHub repo.
+
+The top-level directory comes from openembedded-core and meta-ostro, everything else is in its own
+sub-directory. The ostro-os repository gets updated by importing commits from the individual 
+component repositories. 
+
+For more information about each of these components including the URL,
+branch and current revision used, please refer to the ``conf/combo-layer.conf``
+file in your cloned copy of the top-level ``ostro-os`` repository.
+
+When you initially clone the repository, you clone the aggregation resulting from that script. 
+It is important to note that patches should not go to this aggregated repository; send those to the original layer. 
```

Comment:
However, while developing a fix it can be very convenient to work in the aggregated repository, in particular when patches against different components are necessary. I'm not sure how much we want to go into this. If we do, do we have to explain to less experience developers how to commit them properly (don't mix changes from different components), export patches ("git format-patch"), and apply to individual repos ("git am")?